### PR TITLE
[CU-8689p0gfb] Allows users to unpublish articles

### PR DESCRIPTION
**Pull request by manuelurenah**

### CODE REVIEW

**Comment by JustMaier** [comment]
Task linked: [CU-8689p0gfb Can't unpublish articles](https://app.clickup.com/t/8689p0gfb)

File: src/pages/articles/[id]/[[...slug]].tsx [hunk 1]
```diff
@@ -89,19 +91,43 @@ const MAX_WIDTH = 1320;
...
+  const isModerator = currentUser?.isModerator ?? false;
+  const isOwner = currentUser?.id === article?.user?.id || isModerator;
 
   // boolean value that allows us to disable articles via feature flags and still allow us to show articles created by moderators
   const disableArticles = !articles && !article?.user.isModerator;
 
+  const queryUtils = trpc.useUtils();
+  const upsertArticleMutation = trpc.article.upsert.useMutation();
+  const handlePublishArticle = () => {
+    if (!article || article.status === 'Published') return;
```
Comment (lrojas94) [inline src/pages/articles/[id]/[[...slug]].tsx:113]: use type 👀 ?

File: src/pages/articles/[id]/[[...slug]].tsx [hunk 2]
```diff
@@ -89,19 +91,43 @@ const MAX_WIDTH = 1320;
...
+
+    upsertArticleMutation.mutate(
+      { ...article, status: 'Published' },
```
Comment (lrojas94) [inline src/pages/articles/[id]/[[...slug]].tsx:116]: same

File: src/pages/articles/[id]/[[...slug]].tsx [hunk 3]
```diff
@@ -239,6 +266,15 @@ function ArticleDetailsPage({ id }: InferGetServerSidePropsType<typeof getServer
                 </>
               )}
             </Group>
+            {article.status === 'Unpublished' && isOwner && (
```
Comment (lrojas94) [inline src/pages/articles/[id]/[[...slug]].tsx:269]: same

**Review by lrojas94 — APPROVED** [review]

**Pull request merged**